test/hw9/test_task3.py

```diff
@@ -0,0 +1,58 @@
+import tempfile
+
+import pytest
+
+from All_home_works.hw9.task3 import universal_file_counter
+
+
+@pytest.fixture()
+def make_dir():
+    directory = None
+    files = []
+
+    def _dirname(data):
+        nonlocal directory, files
+        directory = tempfile.TemporaryDirectory()
+        for extension in (".py", ".txt", ".txt"):
+            named_file = tempfile.NamedTemporaryFile(
+                dir=directory.name, suffix=extension, mode="w"
+            )
+            named_file.writelines(data)
+            named_file.seek(0)
+            files.append(named_file)
+        return directory.name
+
+    yield _dirname
+    for file in files:
+        file.close()
+    directory.cleanup()
+
+
+@pytest.mark.parametrize(("extension", "expected"), [("txt", 4), ("py", 2)])
+def test_universal_file_counter__without_tokenizer(make_dir, extension, expected):
+    data = ["one line\n", "second line"]
```

Comment:
It looks strange that this code produces 4 items for txt and 2 for py.    Probably some magic is hidden in `make_dir`.

Could you rewrite this test in a way when it will be absolutely clear why you expect these numbers?

Reply:
This question is still actual.